libs/types/src/investments.rs

```diff
@@ -239,3 +239,59 @@ pub struct ExecutedForeignCollect<Balance, Currency> {
 	///   pool currency)
 	pub amount_remaining: Balance,
 }
+
+/// A representation of an investment portfolio consisting of free, pending and
+/// claimable pool currency as well as tranche tokens.
+#[derive(Encode, Decode, Default, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+
+pub struct InvestmentPortfolio<Balance> {
+	/// The unprocessed invest order amount in pool currency
+	pub pending_invest_currency: Balance,
+	/// The amount of tranche tokens which can be collected for an invest order
+	pub claimable_tranche_tokens: Balance,
+	/// The amount of tranche tokens which can be transferred
+	pub free_tranche_tokens: Balance,
+	/// The amount of tranche tokens which can not be used at all and get
+	/// slashed
+	pub reserved_tranche_tokens: Balance,
```

Comment:
NOTE: Decided against supporting `frozen` as we don't support frozen fungibles in restricted tokens yet and are not freezing anywhere in the codebase.

Reply:
Didn't we want to add the `CurrencyId` of the actual `pool_currency` here as a field?

Reply:
IMO it's impossible to map free `pool_currency` to an `InvestmentId` because
* We will have multiple pools with the same `pool_currency`
* Once it is `free`, the root cause for that is unknown without introducing additional storage

Reply:
Ah sorry. I meant, to add this in order for the UI to show that they user has pending amount of `PoolCurrency` investing. So the UI know what currencie to display. 

Does this make more sense?

Reply:
Sorry, I cant follow. Could you elaborate @mustermeiszer ? We already have a field for pending pool currency. Any pool currency that is not related to an investment id shouldnt be part of this portfolio API per definition (no key value mapping).

Reply:
> `fn investment_portfolio(account_id: AccountId) -> Vec<(InvestmentId, InvestmentPortfolio)>`

as per this API the UI has no knowledge what is the `pool_currency`, so we should give it this information in the structure.

Reply:
Now I get it, you are talking about the _currency id_ while I was still in the _amounts_ space. I will add it then to have a full portfolio without any need for further queries. On that note: The investment id key contains the pool id such that another query suffices to read the pool currency.

Reply:
[60e9add](https://github.com/centrifuge/centrifuge-chain/pull/1608/commits/60e9add23d13fb680cef7fc8f9522bac59e27765)